roles/assemble_template/tasks/main.yml

```diff
@@ -40,9 +40,9 @@
 - name: Create cluster template
   cloudera.cluster.assemble_cluster_template:
     src: "{{ fragments_temp_directory.path }}"
-    dest: "{{ fragments_temp_directory.path }}/cluster-template.json"
-
-- name: Set fact for cluster template path
-  ansible.builtin.set_fact:
-    cluster_template_path: "{{ fragments_temp_directory.path }}/cluster-template.json"
+    dest: "{{ cluster_template_destination_directory | mandatory  }}"
```

Comment:
The `mandatory` variable should be declared in the role's `default/main.yml` file for the `cluster_template_destination_file` variable.